File: Exam/Answers/4/rostring/rostring.c
#include <unistd.h>

void	rostring(char *s)
{
	int x = 0;
	int y = 0;

	while(s[x] == ' ' || s[x] == '\t')
		x++;
	y = x;
	while(s[x] && s[x] != ' ' && s[x] != '\t')
		x++;
	while(s[x])
	{
		if(s[x] && s[x] != ' ' && s[x] != '\t' && (s[x - 1] == ' ' || s[x - 1] == '\t'))
		{
			while(s[x] && s[x] != ' ' && s[x] != '\t')
				write(1, &s[x++], 1);
			write(1, " ", 1);
		}
		x++;
	}
	while(s[y] && s[y] != ' ' && s[y] != '\t')
		write(1, &s[y++], 1);
}

int		main(int argc, char *argv[])
{
	if(argc > 1)
		rostring(argv[1]);
	write(1, "\n", 1);
	return (0);
}

File: PHP_bootcamp/D01/ex07/rostring.php
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    rostring.php                                       :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: cnaude <marvin@42.fr>                      +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2018/10/02 20:50:01 by cnaude            #+#    #+#              #
#    Updated: 2018/10/02 20:50:07 by cnaude           ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

<?php
	if ($argc > 1)
	{
		$string = $argv[1];
		while ((strpos($string, "  ")) == TRUE)
			$string = str_replace("  ", " ", $string);
		$array = explode(" ", $string);
		$wordnbr = count($array);

		for($x = 1; $x < $wordnbr; $x++)
			echo $array[$x]." ";
		echo $array[0]."\n";
	}
?>